### ISSUE TITLE
Another Server Jsonpatch error

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3507]
Version 2.1.1765
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Starcounter.XSON.JsonPatch.EstimateSizeOfPath(Json json, Boolean includeNamespace, Boolean calledFromStepSibling) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 389
   at Starcounter.XSON.JsonPatch.EstimateSizeOfPath(Json json, Boolean includeNamespace, Boolean calledFromStepSibling) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 358
   at Starcounter.XSON.JsonPatch.EstimateSizeOfPath(Json json, Boolean includeNamespace, Boolean calledFromStepSibling) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 349
   at Starcounter.XSON.JsonPatch.EstimateSizeOfPatch(Change change, Boolean includeNamespace) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 292
   at Starcounter.XSON.JsonPatch.Generate(ChangeLog changeLog, Boolean includeNamespace, Boolean flushLog, Byte[]& patches) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 133
   at Starcounter.XSON.JsonPatch.Generate(Json json, Boolean flushLog, Boolean includeNamespace, Byte[]& patches) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 90
   at Starcounter.Internal.JsonMimeConverter.Convert(Request request, IResource before, MimeType mimeType, MimeType& resultingMimeType) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.Apps.JsonPatch\BuiltInRestHandlers\JsonMimeConverter.cs:line 18
   at Starcounter.Page.AsMimeType(MimeType mimeType, MimeType& resultingMimeType, Request request) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter\Page.cs:line 49
   at Starcounter.Response.ConstructFromFields(Request req, Byte[] givenBuffer) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.Internal\Http\Response.cs:line 809
   at Starcounter.Request.SendResponse(Response resp, Byte[] serializationBuf) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.Internal\Http\Request.cs:line 1241
   at Starcounter.Internal.AppsBootstrapper.ProcessExternalRequest(Request req) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-272\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:line 645
HResult=-2147467261
```

**Comment by dan31** [comment]
Thanks @malx122, yet another one from a "null-exception" family. CC @chrhol

**Comment by chrhol** [comment]
The stacktrace is a bit weird, the last line, that the nullreference comes from is this (`totalSize` is an integer):

```csharp
private static int EstimateSizeOfPath(...) {
    ...
    return totalSize; // line 389
}
```

Is there some way to reproduce this issue? The last change in this file is from 2016-03-31, so it's not something new that have been added that makes the linenumbers/sourcecode different from your version.

**Comment by malx122** [comment]
I have discussed this with Christian.
There is the SignInApp that removes dbobject `UserSession` and another app that has a commit hook for that object. In that commit hook the other app modifies (removes) som of its Json properties currently being displayed, but make no effort of pushing or anything like that. 

I have worked around it and no longer use commit hook for this purpose, mostly because the workaround probably was a better solution. (So the bug does not affect us any longer even though it remains)

**Comment by chrhol** [comment]
I think I got this exception when playing around with the Warehouse just now. Will investigate.

**Comment by chrhol** [comment]
I found a consistent way of reproducing this issue (or at least I think it's the same issue) using the warehouse apps. 

It's a bit hard to tell since the stacktrace in the first post is a bit off. And to be honest I'm not quite sure what actually happens, since it involves a lot of different things happening in some specific order. I also tried to come up with some way to write a unittest or regressiontest for this, but without success there.

What I can say that after my change, there is no longer any nullreferenceexception and the patches seems to be generated correctly. So without anything more to go on, I will simply consider this issue solved...

I will push the fix to develop and pnext soon.
